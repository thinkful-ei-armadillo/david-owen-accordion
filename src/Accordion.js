import React, {Component} from 'react';
import BigMenu from './BigMenu';
import SmallMenu from './SmallMenu';


class Accordion extends Component {
  state = {
    clicked: null

  }

  toggleClicked = (id) => {
    this.setState({clicked: id})
  }

  render () {
    console.log(this.state);
    return (
      <ul className="menus">
      {this.props.sections.map((item, index) => (index === this.state.clicked) ? <BigMenu key={index} id={index} title={item.title} content={item.content} toggle={this.toggleClicked} /> : <SmallMenu key={index} id={index} title={item.title} toggle={this.toggleClicked} />)}
      </ul>
    )
  }
}



export default Accordion;