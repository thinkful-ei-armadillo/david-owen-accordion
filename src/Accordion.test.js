import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Accordion from './Accordion';

describe('snapshot testing accordion', () => {
it('should be empty when no sections is supplied', ()=>{
  const testSections = [{
    title: 'section 1',
    content: 'E nomine'
  }]
  const tree  = renderer
  .create(<Accordion sections={null} />)
  .toJSON();
  expect(<Accordion sections={null}/>).toBe(
    <ul className="menus"></ul>
  );
});
it('should render no sections as active (BigMenu components) by default ', ()=>{
  const testSections =
});