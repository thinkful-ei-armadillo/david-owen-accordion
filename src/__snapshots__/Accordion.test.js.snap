// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the menus as expected 1`] = `
<ul
  className="menus"
>
  <li
    id={0}
  >
    <button
      onClick={[Function]}
    >
      section 1
    </button>
  </li>
</ul>
`;
